Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.API/Controllers/TimeOffController.cs

```diff
@@ -244,25 +244,38 @@ internal async Task ProcessTimeOffsAsync(string isRequestFromLogicApp)
             List<PayCodeToTimeOffReasonsMappingEntity> timeOffRequestsPayCodeList,
             List<GlobalTimeOffRequestItem> globalTimeOffRequestDetails)
         {
+            this.telemetryClient.TrackTrace($"ProcessTimeOffEntitiesBatchAsync start at: {DateTime.UtcNow.ToString("O", CultureInfo.InvariantCulture)}");
+
             foreach (var item in processKronosUsersQueueInBatch)
             {
                 foreach (var timeOffRequestItem in timeOffResponseDetails.Where(x => x.Employee.PersonIdentity.PersonNumber == item.KronosPersonNumber))
                 {
                     if (timeOffRequestItem.StatusName.Equals(ApiConstants.ApprovedStatus, StringComparison.Ordinal))
                     {
+                        this.telemetryClient.TrackTrace($"ProcessTimeOffEntitiesBatchAsync look up count: {lookUpData.Count} ");
+
                         if (lookUpData.Count == 0)
                         {
                             var timeOffReasonId = timeOffReasons.
                                 Where(t => t.RowKey == timeOffRequestItem.TimeOffPeriods.TimeOffPeriod.PayCodeName && t.PartitionKey == item.ShiftTeamId).FirstOrDefault();
-                            timeOffNotFoundList.Add(timeOffRequestItem);
-                            userModelNotFoundList.Add(item);
-                            kronosPayCodeList.Add(timeOffReasonId);
+                            this.telemetryClient.TrackTrace($"ProcessTimeOffEntitiesBatchAsync PaycodeName : {timeOffRequestItem.TimeOffPeriods.TimeOffPeriod.PayCodeName} ");
+                            this.telemetryClient.TrackTrace($"ProcessTimeOffEntitiesBatchAsync ReqId : {timeOffRequestItem.Id} ");
+                            if (timeOffReasonId != null)
```

Comment:
This needs comments: When is timeOffReasonId null? Is that expected? A configuration error? Why is it safe to skip those entities?

Reply:
@aosolis - ack'd. Will be adding the necessary comments - all of the above will be resolved in the next commit that is made on this PR.